lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1326,6 +1331,21 @@ DIFile *SPIRVToLLVMDbgTran::getFile(const SPIRVId SourceId) {
   return getDIFile(getString(SourceArgs[FileIdx]), ParseChecksum(ChecksumStr));
 }
 
+DIBuilder &SPIRVToLLVMDbgTran::getDIBuilder(const SPIRVExtInst *DebugInst) {
+  assert(BuilderMap.size() != 0 && "No debug compile units");
+  if (BuilderMap.size() == 1)
+    return *BuilderMap.begin()->second;
+  while (DebugInst->getExtOp() != SPIRVDebug::CompilationUnit) {
+    if (DebugInst->getExtOp() == SPIRVDebug::DebugInfoNone)
+      return *BuilderMap.begin()->second;
+    if (SPIRVEntry *Scope = DebugInst->getDebugScope())
+      DebugInst = static_cast<SPIRVExtInst *>(Scope);
+    else
+      return *BuilderMap.begin()->second;
```

Comment:
```suggestion
    if (SPIRVEntry *Scope = DebugInst->getDebugScope()) {
      DebugInst = static_cast<SPIRVExtInst *>(Scope);
      continue;
    }
    return *BuilderMap.begin()->second;
```
or something like this